// @lifi/sdk/src/_cjs/core/EVM/abi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.eip2612Types = exports.balanceOfAbi = exports.getEthBalanceAbi = exports.allowanceAbi = exports.approveAbi = exports.eip2612Abi = exports.permit2ProxyAbi = void 0;
const viem_1 = require("viem");
exports.permit2ProxyAbi = (0, viem_1.parseAbi)([
    'function callDiamondWithPermit2(bytes, ((address, uint256), uint256, uint256), bytes) external',
    'function callDiamondWithEIP2612Signature(address, uint256, uint256, uint8, bytes32, bytes32, bytes) external payable',
    'function nextNonce(address) external view returns (uint256)',
    'function callDiamondWithPermit2Witness(bytes, address, ((address, uint256), uint256, uint256), bytes) external payable',
]);
exports.eip2612Abi = (0, viem_1.parseAbi)([
    'function permit(address, address, uint256, uint256, uint8, bytes32, bytes32) external',
    'function DOMAIN_SEPARATOR() external view returns (bytes32)',
    'function nonces(address) external view returns (uint256)',
    'function name() external view returns (string)',
    'function version() external view returns (string)',
]);
exports.approveAbi = (0, viem_1.parseAbi)([
    'function approve(address, uint256) external returns (bool)',
]);
exports.allowanceAbi = (0, viem_1.parseAbi)([
    'function allowance(address, address) external view returns (uint256)',
]);
exports.getEthBalanceAbi = (0, viem_1.parseAbi)([
    'function getEthBalance(address) external view returns (uint256)',
]);
exports.balanceOfAbi = (0, viem_1.parseAbi)([
    'function balanceOf(address) external view returns (uint256)',
]);
exports.eip2612Types = {
    Permit: [
        { name: 'owner', type: 'address' },
        { name: 'spender', type: 'address' },
        { name: 'value', type: 'uint256' },
        { name: 'nonce', type: 'uint256' },
        { name: 'deadline', type: 'uint256' },
    ],
};
