// @bigmi/core/dist/cjs/clients/decorators/public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.publicActions = publicActions;
const getBalance_js_1 = require("../../actions/getBalance.js");
const getBlock_js_1 = require("../../actions/getBlock.js");
const getBlockCount_js_1 = require("../../actions/getBlockCount.js");
const getBlockStats_js_1 = require("../../actions/getBlockStats.js");
const getTransactions_js_1 = require("../../actions/getTransactions.js");
const getUTXOTransaction_js_1 = require("../../actions/getUTXOTransaction.js");
const getUTXOs_js_1 = require("../../actions/getUTXOs.js");
function publicActions(client) {
    return {
        getBalance: (args) => (0, getBalance_js_1.getBalance)(client, args),
        getBlock: (args) => (0, getBlock_js_1.getBlock)(client, args),
        getBlockCount: () => (0, getBlockCount_js_1.getBlockCount)(client),
        getBlockStats: (args) => (0, getBlockStats_js_1.getBlockStats)(client, args),
        getTransactions: (args) => (0, getTransactions_js_1.getTransactions)(client, args),
        getUTXOTransaction: (args) => (0, getUTXOTransaction_js_1.getUTXOTransaction)(client, args),
        getUTXOs: (args) => (0, getUTXOs_js_1.getUTXOs)(client, args),
    };
}
