// @lifi/sdk/src/_cjs/core/Sui/Sui.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Sui = Sui;
const types_1 = require("@lifi/types");
const utils_1 = require("@mysten/sui/utils");
const SuiStepExecutor_js_1 = require("./SuiStepExecutor.js");
const getSuiBalance_js_1 = require("./getSuiBalance.js");
const getSuiNSAddress_js_1 = require("./getSuiNSAddress.js");
function Sui(options) {
    const _options = options ?? {};
    return {
        get type() {
            return types_1.ChainType.MVM;
        },
        isAddress: utils_1.isValidSuiAddress,
        resolveAddress: getSuiNSAddress_js_1.getSuiNSAddress,
        getBalance: getSuiBalance_js_1.getSuiBalance,
        async getStepExecutor(options) {
            if (!_options.getWallet) {
                throw new Error('getWallet is not provided.');
            }
            const wallet = await _options.getWallet();
            const executor = new SuiStepExecutor_js_1.SuiStepExecutor({
                wallet,
                routeId: options.routeId,
                executionOptions: {
                    ...options.executionOptions,
                },
            });
            return executor;
        },
        setOptions(options) {
            Object.assign(_options, options);
        },
    };
}
